frontend/public/components/build.tsx

```diff
@@ -127,40 +128,52 @@ export const BuildNumberLink = ({ build }) => {
 };
 
 const BuildGraphs = requirePrometheus(({ build }) => {
-  const podName = _.get(build, ['metadata', 'annotations', 'openshift.io/build.pod-name']);
+  const podName = build.metadata.annotations?.['openshift.io/build.pod-name'];
   if (!podName) {
     return null;
   }
-
+  const endTime = build?.status?.completionTimestamp
+    ? new Date(build.status.completionTimestamp).getTime()
+    : Date.now();
+  const runTime = build?.status?.startTimestamp
+    ? endTime - new Date(build.status.startTimestamp).getTime()
+    : ONE_HOUR;
+  const timespan = Math.max(runTime, ONE_MINUTE); // Minimum timespan of one minute
   const namespace = build.metadata.namespace;
-
+  const areaProps = {
+    namespace,
+    endTime,
+    timespan,
+    formatDate: (d) => twentyFourHourTime(d, timespan < 5 * ONE_MINUTE),
+    domain: { x: [endTime - timespan, endTime] }, // force domain to match queried timespan
```

Comment:
I think we're OK in this case, but in general I've had trouble with triggering unnecessary graph re-renders by having function / object / array props that are recreated in the render function like this.

If the graph does not seem to be re-rendering too frequently, I think this is fine, but in general I'm trying to avoid these situations. For example, passing `startTime` and `endTime` integer props instead of a `domain` object might avoid some re-renders, depending on how `BuildGraphs` is used.

Reply:
I'll pull out `domain` into a separate declaration and memoize it so it will only get re-initialized if `timespan` or `endTime` change.

Reply:
I'll also memoize `areaProps` so that it won't cause unnecessary re-renders.